### ISSUE TITLE
workflows errors

### ISSUE THREAD

**Issue by plastovicka** [SabreDevStudio/postman-collections#3]
I tried to run Sabre APIs workflows, but all failed.

1 - Air Shop, Price Check, Book, Fulfill
ERR.SP.CLIENT.INVALID_REQUEST - Passive air segments cannot be booked together with regular/active air segments

2 - Air Shop, Price Check, Book, Fulfill (upsell with Brands)
WARN.SWS.HOST.ERROR_IN_RESPONSE - EnhancedAirBookRQ: INVALID EXPIRATION DATE-0092

3 - Air Shop, Price Check, Book, Fulfill  (upsell with Flex Fares)
WARN.SP.HALT_ON_STATUS_RECEIVED - EnhancedAirBookRQ: Specified HaltOnStatus Received - Processing Aborted

4 - Air Shop, Book, Fulfill, Cancel, Void
ERR.2SG.SERVICE_VERSION_DEPRECATED - Requested service STPS_OFFERSSHOPAPI version v2 is deprecated

5 - Air NDC Shop, Price Check, Book, Fulfill
ERR.2SG.SERVICE_VERSION_DEPRECATED - Requested service STPS_OFFERSSHOPAPI version v2 is deprecated

6 - Air LCC Shop, Book, Cancel
ERR.2SG.SERVICE_VERSION_DEPRECATED - Requested service STPS_OFFERSSHOPAPI version v2 is deprecated

7 - CSL: Shop, Price Check, Book -Create PNR
WARN.0102 - No rate data Available

8 - CSL: Shop, Price Check, Book -Update PNR
ERR.SP.PROVIDER_ERROR - PNR has not been updated successfully, see remaining messages for details
WARN.SP.INTERNAL_ERROR - CreatePassengerNameRecordRQ: After maximum retry count it was not possible to successfully execute target service

9 - Vehicle: Shop, Book -Create PNR
ERR.SP.PROVIDER_ERROR - Unable to perform vehicle booking. Please see following messages for details
WARN.SWS.HOST.ERROR_IN_RESPONSE - OTA_VehResLLSRQ: INVALID EXPIRATION DATE

**Comment by tomaszdrag** [comment]
Hi,
Sorry for the long wait - I actually did not receive a notification of your comments. But... here are my responses.

TL:DR I have just created a new version of that collection (Sabre APIs 2022.11) - feel free to check.

1 - Air Shop, Price Check, Book, Fulfill
ERR.SP.CLIENT.INVALID_REQUEST - Passive air segments cannot be booked together with regular/active air segments

Just checked and was able to successfully complete the e2e flow.  The error you mentioned points at step '3' and the .OriginDestinationInformation object. Did you run this unchanged? 

2. 2 - Air Shop, Price Check, Book, Fulfill (upsell with Brands)
WARN.SWS.HOST.ERROR_IN_RESPONSE - EnhancedAirBookRQ: INVALID EXPIRATION DATE-0092

This is due to the expired date of the credit card. Changed it. Also I updated the request to '3' step as the flight segments contained incorrect booking classes ('ResBookDesigCode')

3 - Air Shop, Price Check, Book, Fulfill (upsell with Flex Fares)
WARN.SP.HALT_ON_STATUS_RECEIVED - EnhancedAirBookRQ: Specified HaltOnStatus Received - Processing Aborted

This happens as the value of flight booking class ('ResBookDesigCode') is not parametrized. Check the value returned in step (2.) and pass it correctly in step (3.)

4 - Air Shop, Book, Fulfill, Cancel, Void
ERR.2SG.SERVICE_VERSION_DEPRECATED - Requested service STPS_OFFERSSHOPAPI version v2 is deprecated

Updated the service version and this works correctly. I also updated some 'shop' test cases as they did not work properly for NDC

5 - Air NDC Shop, Price Check, Book, Fulfill
ERR.2SG.SERVICE_VERSION_DEPRECATED - Requested service STPS_OFFERSSHOPAPI version v2 is deprecated

Updated the service version and this works correctly. I also updated some 'shop' test cases as they did not work properly for NDC

6 - Air LCC Shop, Book, Cancel
ERR.2SG.SERVICE_VERSION_DEPRECATED - Requested service STPS_OFFERSSHOPAPI version v2 is deprecated

Updated the service version and this works correctly. I also updated some 'shop' test cases as they did not work properly for NDC

7 - CSL: Shop, Price Check, Book -Create PNR
WARN.0102 - No rate data Available

Ran it and it works OK - probably the hotel rate was not available at the time you checked. Hotel suppliers are picky with their availability data.

8 - CSL: Shop, Price Check, Book -Update PNR
ERR.SP.PROVIDER_ERROR - PNR has not been updated successfully, see remaining messages for details
WARN.SP.INTERNAL_ERROR - CreatePassengerNameRecordRQ: After maximum retry count it was not possible to successfully execute target service

This error was due to a lack of response coming from a reservation downline service. I have added minor changes and it runs ok

9 - Vehicle: Shop, Book -Create PNR
ERR.SP.PROVIDER_ERROR - Unable to perform vehicle booking. Please see following messages for details
WARN.SWS.HOST.ERROR_IN_RESPONSE - OTA_VehResLLSRQ: INVALID EXPIRATION DATE

Updated credit card expiration date as well as some other minor things. Runs ok now.

FYI Remember to set up ticket printers accordingly for the step (4.Fulfil) to succeed. You can contact your account manager for help on this matter.

**Comment by tomaszdrag** [comment]
Since it's been quiet i'll close this issue. If problems occur please open a new one.
